### DISCUSSION TITLE
Not connected to pub400.com! All configured authentication methods failed

### DISCUSSION THREAD

**Discussion by faouzi-ai** [codefori/vscode-ibmi#1280] [General]
Hello 
I just purchased a new windows PC along aside my old Macbook. :-)

   - Configure SSH at port 2222 and Connect  to pub400.com with Putty is OK.
   - Configure TN5250 and connect to pub400.com is OK

I tried to have a connection with  VS code but failed and get the message : Not connected to pub400.com! All configured authentication methods failed

here is the content of the file settings.json: 
`{
"redhat.telemetry.enabled": true,
    "code-for-ibmi.connections": [
        {
            "name": "IBMi-PUB400",
            "host": "pub400.com",
            "port": 2222,
            "username": "FABBES",
            "privateKey": ""
        }
    ],
    "code-for-ibmi.connectionSettings": [
        
    ],
    "rsp-ui.enableStartServerOnActivation": [
        {
            "id": "redhat.vscode-community-server-connector",
            "name": "Community Server Connector",
            "startOnActivation": true9
        }
    ],
    "workbench.colorTheme": "Visual Studio Light"
}`

1.  I can't figure it out, any help please !
 
2  Even I don't use Public/Private Keys, I use password,  the element  at line 9   specify :  PrivateKey.  I don't know why and how .
   I tried twice to avoid this kind of configuration, but still have PrivateKey at line 9. Did I miss something somewhere else ?

**Comment by sebjulliand** [comment]
Hi @faouzi-ai !
No worries, the `privateKey` field is blank and that's normal. Mine is too and I can connect to PUB400.
The only way I could see this error message was to provide the wrong password. Hopefully there's no issue with the way passwords are stored.

Could you check the following points please?

- First, the obvious: update you login settings and put your password here again. We never know, maybe a whitespace made its way at the end of it if it was copy/pasted or there may be an error in it.
- If the password is OK but it still fails to connect, create another connection to pub400 without setting any password. Try to connect and you'll be prompted for your password. See if it works this time when provided through the prompt.
- If it does, then there may be an issue with the way the password it retrieved from the login editor in which case this will require further investigations.

Are there very special characters (i.e. specific to a language in particular) in your password?

**Comment by faouzi-ai** [comment]
Thank you Seb

Your last clue triggered the solution in my mind
Although I am used to working in an international context, I sometimes forget that I have an AZERTY keyboard
